### PR TITLE
Make `sqw@smooth` return `dnd`

### LINKED ISSUE #1583: smooth breaks consistency between pixels and image
smooth algorithm works well on `dnd` object making image more pleasent to eye. 
When it is applied to `sqw` object, it chagnes image only so `recompute_bin_data` would return different result

This is not good as by default it is assumed that pixels and image are consistent between each other. 
`smooth` should be either prohibited on `sqw` or it should change `PixelData` to be consistent with image.

### PR DESCRIPTION
**Pull request by oerc0122**
Fixes #1583

### CODE REVIEW

File: horace_core/sqw/@sqw/smooth.m [hunk 1]
```diff
@@ -1,4 +1,4 @@
-function wout = smooth(win, varargin)
+function wout_dnd = smooth(win, varargin)
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@sqw/smooth.m:1]: not part of your change but next line presuming gataway is gateway, not getaway. Having said which, I still don't know what it means

be good to have explicit description of the input and output arguments, so
input win - sqw without pixels. Although is this still a requirement now the output is a guaranteed dnd separate from any pixel data?
input varargin - parameters for the smooth method for dnds
output wout_dnd - dnd object with image pixels smoothed with respect to the image pixels in input win

File: horace_core/sqw/@sqw/smooth.m [hunk 2]
```diff
@@ -14,8 +14,6 @@
 
 % return cellarray of corresponding dnd objects
 wout_dnd = dnd(win,'-cell');
-wout = copy(win);
+wout_dnd = cellfun(@(x) smooth(x, varargin{:}), wout_dnd);
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@sqw/smooth.m:17]: precede with
% apply dnd smooth method to dnd conversion of input win

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]

File: horace_core/sqw/@sqw/smooth.m [hunk 3]
```diff
@@ -1,21 +1,49 @@
...
+%           If false, unit of measure is pixels
+%
+%   width   Vector that sets the extent of the smoothing along each dimension.
+%           The interpretation of width depends on the argument 'shape' described
+%           below.
+%
+%           If width is scalar, then the value is applied to all dimensions
+%
+%           e.g. if win.data is a d3d, valid arguments for width might be:
+%                width = [2,4,5]    % 2, 3, 5 along the 1st, 2nd and 3rd dimensions
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@sqw/smooth.m:20]: make value and comment consistent

File: _test/test_sqw_class/test_smooth.m [hunk 4]
```diff
@@ -13,161 +12,115 @@
...
-                'HORACE:sqw:invalid_argument');
-            assertTrue(contains(actual.message, 'No smoothing of sqw data'))
-        end
-
         function test_smooth_raises_error_with_wrong_dimension_width_arg(obj)
-            s = sqw(obj.test_sqw_2d_fullpath);
-            s_nopix = copy(s);
-            s_nopix.pix = PixelDataBase.create();
+            sqw_in = sqw(obj.test_sqw_2d_fullpath);
+            sqw_in_nopix = copy(sqw_in);
```
Comment (cmarooney-stfc) [inline _test/test_sqw_class/test_smooth.m:117]:
```suggestion
            sqw_in_nopix = copy(sqw_in); % nopix anticipates later depixelation
```

File: horace_core/sqw/@sqw/smooth.m (hunk 3 above)
Comment (oerc0122) [reply horace_core/sqw/@sqw/smooth.m:20]:
```suggestion
%                width = [2,4,5]    % 2, 4, 5 along the 1st, 2nd and 3rd dimensions
```

**Pull request merged**